packages/types/src/scripts/srmlTS.ts

```diff
@@ -6,25 +6,30 @@ import fs from 'fs';
 import { isString } from '@polkadot/util';
 
 import { getTypeDef, TypeDef, TypeDefInfo, TypeDefExtVecFixed } from '../codec/createType';
+import * as primitives from '../primitive';
+import * as substrate from '../type';
 import * as definitions from '../srml/definitions';
 
 // these map all the codec and primitive types for import, see the TypeImports below. If
 // we have an unseen type, it is `undefined`/`false`, if we need to import it, it is `true`
-interface TypeExist {
-  [index: string]: boolean;
-}
+type TypeExist = Record<string, boolean>;
 interface TypeImports {
   codecTypes: TypeExist;
-  otherTypes: TypeExist;
   ownTypes: string[];
+  primitiveTypes: TypeExist;
+  substrateTypes: TypeExist;
 }
 
 const HEADER = '/* eslint-disable @typescript-eslint/no-empty-interface */\n// Auto-generated, do not edit\n\n';
 const FOOTER = '\n';
 
-function setImports ({ codecTypes, otherTypes, ownTypes }: TypeImports, type: string | null, codecType: string | null): void {
-  if (type && !ownTypes.includes(type) && !otherTypes[type]) {
-    otherTypes[type] = true;
+function setImports ({ codecTypes, ownTypes, primitiveTypes, substrateTypes }: TypeImports, type: string | null, codecType: string | null): void {
```

Comment:
Make is slightly more clever - it can now import from `types/type` in addition to `types/primitive`